lib/engine/game/g_1880/step/buy_train.rb

```diff
@@ -80,8 +80,7 @@ def owns_restored2?(entity)
 
           def can_buy_restored2?(entity)
             @game.phase.available?(restored2_train&.available_on) &&
-            !owns_restored2?(entity) &&
-            @corporations_sold.empty?
```

Comment:
This is incorrect. The president cannot contribute towards purchasing a restored 2T.